### PR TITLE
Talk about git submodules

### PR DESCRIPTION
**Pull request by LeSeulArtichaut**
Helps with #850.
cc @jyn514

### CODE REVIEW

File: src/git.md [hunk 1]
```diff
@@ -73,6 +73,30 @@ the same, with some steps skipped:
...
+	modified:   src/tools/cargo (new commits)
+	modified:   src/tools/rls (new commits)
+	modified:   src/tools/rustfmt (new commits)
+
+no changes added to commit (use "git add" and/or "git commit -a")
+```
+
+These changes are not changes to files: they are changes to submodules
+(more on this later). To get rid of those, run `git submodule update` (or run any
+`x.py` command which will automatically update the submodules).
```
Comment (jyn514) [inline src/git.md:98]:
```suggestion
`x.py` command, which will automatically update the submodules).
```
Before it sounded like you had to know which commands would update the modules, but in reality any command will work.

File: src/git.md [hunk 2]
```diff
@@ -73,6 +73,30 @@ the same, with some steps skipped:
...
+
```
Comment (jyn514) [inline src/git.md:99]: Github suggestions don't like nested triple quotes, but can you add this?

Note that there is currently a bug if you use `x.py`, worktrees, and submodules. If you run into an error like
```
error: failed to read `/home/joshua/rustc-worktree/src/tools/miri/cargo-miri/Cargo.toml`

Caused by:
  No such file or directory (os error 2)
```
it's not anything you did wrong. There is a workaround at [#77620](https://github.com/rust-lang/rust/issues/77620#issuecomment-705228229).

**Review by jyn514 — COMMENTED** [review]
cc @thomcc - does this make you a little more comfortable with submodules?

File: src/git.md (hunk 2 above)
Comment (camelid) [reply src/git.md:99]: @jyn514 Something I learned: you _can_ nest triple backticks if you use triple tildes as the outer delimiter:

~~~suggestion
```
foo
```
~~~

^ just for demonstration, don't apply!

Comment (jyn514) [reply src/git.md:99]: Works great, thanks!

Comment (LeSeulArtichaut) [reply src/git.md:102]: :o That's super helful, thank you ❤️

**Review by jyn514 — APPROVED** [review]
LGTM, we can make improvements in follow-ups.

**Pull request merged**